testing/json.test

```diff
@@ -140,11 +180,178 @@ do_execsql_test json_extract_null_path {
     SELECT json_extract(1, null)
 } {{}}
 
+do_execsql_test json_arrow_null_path {
+    SELECT 1 -> null
+} {{}}
+
+do_execsql_test json_arrow_shift_null_path {
+    SELECT 1 ->> null
+} {{}}
+
+do_execsql_test json_extract_float {
+  SELECT typeof(json_extract(1.0, '$'))
+} {{real}}
+
+do_execsql_test json_arrow_float {
+  SELECT typeof(1.0 -> '$')
+} {{text}}
+
+do_execsql_test json_arrow_shift_float {
+  SELECT typeof(1.0 ->> '$')
+} {{real}}
+
+do_execsql_test json_extract_true {
+  SELECT json_extract('true', '$')
+} {{1}}
+
+do_execsql_test json_extract_true_type {
+  SELECT typeof(json_extract('true', '$'))
+} {{integer}}
+
+do_execsql_test json_arrow_true {
+  SELECT 'true' -> '$'
+} {{true}}
+
+do_execsql_test json_arrow_true_type {
+  SELECT typeof('true' -> '$')
+} {{text}}
+
+do_execsql_test json_arrow_shift_true {
+  SELECT 'true' ->> '$'
+} {{1}}
+
+do_execsql_test json_arrow_shift_true_type {
+  SELECT typeof('true' ->> '$')
+} {{integer}}
+
+do_execsql_test json_extract_false {
+  SELECT json_extract('false', '$')
+} {{0}}
+
+do_execsql_test json_extract_false_type {
+  SELECT typeof(json_extract('false', '$'))
+} {{integer}}
+
+do_execsql_test json_arrow_false {
+  SELECT 'false' -> '$'
+} {{false}}
+
+do_execsql_test json_arrow_false_type {
+  SELECT typeof('false' -> '$')
+} {{text}}
+
+do_execsql_test json_arrow_shift_false {
+  SELECT 'false' ->> '$'
+} {{0}}
+
+do_execsql_test json_arrow_shift_false_type {
+  SELECT typeof('false' ->> '$')
+} {{integer}}
+
+do_execsql_test json_extract_string {
+  SELECT json_extract('"string"', '$')
+} {{string}}
+
+do_execsql_test json_extract_string_type {
+  SELECT typeof(json_extract('"string"', '$'))
+} {{text}}
+
+do_execsql_test json_arrow_string {
+  SELECT '"string"' -> '$'
+} {{"string"}}
+
+do_execsql_test json_arrow_string_type {
+  SELECT typeof('"string"' -> '$')
+} {{text}}
+
+do_execsql_test json_arrow_shift_string {
+  SELECT '"string"' ->> '$'
+} {{string}}
+
+do_execsql_test json_arrow_shift_string_type {
+  SELECT typeof('"string"' ->> '$')
+} {{text}}
+
+do_execsql_test json_arrow_implicit_root_path {
+  SELECT '{"a":1}' -> 'a';
+} {{1}}
+
+do_execsql_test json_arrow_shift_implicit_root_path {
+  SELECT '{"a":1}' ->> 'a';
+} {{1}}
+
+# TODO: fix me after rebasing on top of https://github.com/tursodatabase/limbo/pull/631 - use the Option value in json_extract_single
+#do_execsql_test json_arrow_implicit_root_path_undefined_key {
+#  SELECT '{"a":1}' -> 'x';
+#} {{}}
+
+do_execsql_test json_arrow_shift_implicit_root_path_undefined_key {
+  SELECT '{"a":1}' ->> 'x';
+} {{}}
+
+do_execsql_test json_arrow_implicit_root_path_array {
+  SELECT '[1,2,3]' -> 1;
+} {{2}}
+
+do_execsql_test json_arrow_shift_implicit_root_path_array {
+  SELECT '[1,2,3]' ->> 1;
+} {{2}}
+
+do_execsql_test json_arrow_implicit_root_path_array_negative_idx {
```

Comment:
This test fails with the SQLite used on CI but works with the newer SQLite `3.47.2`:

```
SQLite version 3.47.2 2024-12-07 20:39:59
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> SELECT '[1,2,3]' -> -1;
3
```

Should we install the newer one on CI?